docs/_docs/cross-chain/core-messaging-layer.md

```diff
@@ -0,0 +1,59 @@
+---
+---
+# Obscuro Cross Chain Messaging
+
+Obscuro is equipped with a cross chain messaging protocol that enables sending and receiving messages across layers securely and in an entirely decentralized fashion.
+
+The core contract that provides this functionality is the `MessageBus`. It exists both on L1 and L2. In the L1 you can find it under the management contract whilst on the L2 it is created and managed by the enclave.
+
+## How It Works
+
+Users call the `publishMessage` function on the `MessageBus`. This function emits an event that records what message is being published, who is calling the method and some other bits. This event later on gets picked on by the enclave which ensures that the message is available on the counterpart `MessageBus`.
```

Comment:
Is it worth detailing how it gets picked up i.e. (if my understanding is correct) the enclave consume the blocks from the L1 and reads the message as an event within the transaction logs?

Reply:
I've tried to avoid in depth detail as I think it might be better suited inside the whitepaper/tech design (or another docs section really).